api/provider/aws/service_create.go

```diff
@@ -77,17 +110,33 @@ func (s *ServiceProvider) Create(req models.CreateServiceRequest) (string, error
 
 func (s *ServiceProvider) createService(
 	cluster string,
+	launchType string,
 	serviceName string,
 	taskDefinition string,
 	desiredCount int,
 	loadBalancerRole string,
 	loadBalancer *ecs.LoadBalancer,
+	securityGroupIDs []*string,
+	subnets []*string,
 ) error {
 	input := &ecs.CreateServiceInput{}
 	input.SetCluster(cluster)
 	input.SetDesiredCount(int64(desiredCount))
+	input.SetLaunchType(launchType)
 	input.SetServiceName(serviceName)
 	input.SetTaskDefinition(taskDefinition)
+
+	if launchType == ecs.LaunchTypeFargate {
+		awsvpcConfig := &ecs.AwsVpcConfiguration{}
+		awsvpcConfig.SetAssignPublicIp(ecs.AssignPublicIpDisabled)
+		awsvpcConfig.SetSecurityGroups(securityGroupIDs)
+		awsvpcConfig.SetSubnets(subnets)
+		networkConfig := &ecs.NetworkConfiguration{}
+		networkConfig.SetAwsvpcConfiguration(awsvpcConfig)
+		input.SetNetworkConfiguration(networkConfig)
+		input.SetPlatformVersion("LATEST")
```

Comment:
[nit] can't help but call out whenever I see "latest" vs a hardcoded version; is this the best option here?

Reply:
At the moment, it seems like the only platform version available is `"1.0.0"`, and that `"LATEST"` is currently synonymous; using "LATEST" here was just a holdover from following guides and just getting something to work; it's probably safer to manage this value ourselves, and I think it's probably a good idea to set this in the config somewhere instead of setting it here.

https://docs.aws.amazon.com/AmazonECS/latest/developerguide/platform_versions.html

Reply:
✅